services/web/server/src/simcore_service_webserver/resource_manager/garbage_collector.py

```diff
@@ -73,9 +81,53 @@ async def collect_garbage(registry: RedisResourceRegistry, app: web.Application)
                 )
 
                 await remove_resources_if_guest_user(
-                    app=app, project_uuid=resource_value, user_id=int(key["user_id"])
+                    app=app,
+                    project_uuid=resource_value,
+                    user_id=int(dead_key["user_id"]),
                 )
 
+    # remove possible pending contianers
+    await remove_orphaned_services(registry, app)
+
+
+async def remove_orphaned_services(
+    registry: RedisResourceRegistry, app: web.Application
+) -> None:
+    """Removes services which are no longer tracked in the database
+
+    Multiple deployments can be active at the same tim on the same cluster.
+    This will also check the current SWARM_STACK_NAME of the service which
+    must be matching its own.
```

Comment:
maybe state that the director takes care of it?

Reply:
added more to the description